mesonbuild/interpreter.py

```diff
@@ -2065,6 +2091,121 @@ def get_cross_property_method(self, args, kwargs):
                     }
 
 
+class DependencyFallbacksHolder(InterpreterObject):
+    def __init__(self, interp):
+        InterpreterObject.__init__(self)
+        self.methods.update({'has_function': self.has_function_method,
+                             'find_library': self.find_library_method,
+                             'dependency': self.dependency_method,
+                             'subproject': self.subproject_method,
+                            })
+        self.interpreter = interp
+        self.fallbacks = []
+
+        wrap_mode = self.interpreter.coredata.get_builtin_option('wrap_mode')
+        self.nofallback = wrap_mode == WrapMode.nofallback
+        self.forcefallback = wrap_mode == WrapMode.forcefallback
+
+    @FeatureNew('dependency_fallbacks.has_function', '0.53.0')
+    @permittedKwargs(has_function_permitted_kwargs)
+    def has_function_method(self, args, kwargs):
+        self.interpreter.validate_arguments(args, 2, [CompilerHolder, str])
+        self.fallbacks.append((self._do_has_function, args, kwargs))
+
+    @FeatureNew('dependency_fallbacks.find_library', '0.53.0')
+    @permittedKwargs(find_library_permitted_kwargs - {'required'})
+    def find_library_method(self, args, kwargs):
+        self.interpreter.validate_arguments(args, 2, [CompilerHolder, str])
+        self.fallbacks.append((self._do_find_library, args, kwargs))
+
+    @FeatureNew('dependency_fallbacks.dependency', '0.53.0')
+    @permittedKwargs(permitted_kwargs['dependency'] - {'required', 'fallback'})
+    def dependency_method(self, args, kwargs):
+        self.interpreter.validate_arguments(args, 1, [str])
+        self.fallbacks.append((self._do_dependency, args, kwargs))
+
+    @FeatureNew('dependency_fallbacks.subproject', '0.53.0')
+    @permittedKwargs(permitted_kwargs['subproject'] - {'required'} | {'variable'})
+    def subproject_method(self, args, kwargs):
+        self.interpreter.validate_arguments(args, 1, [str])
+        self.fallbacks.append((self._do_subproject, args, kwargs))
+
+    def _do_has_function(self, name, display_name, kwargs, func_args, func_kwargs):
+        compiler = func_args[0]
+        del func_kwargs['required']
+        if compiler.has_function_method(func_args[1:], func_kwargs):
+            dep = dependencies.InternalDependency('undefined', [], [], [], [], [], [], [])
+            return DependencyHolder(dep, self.interpreter.subproject)
+        return None
+
+    def _do_find_library(self, name, display_name, kwargs, func_args, func_kwargs):
+        compiler = func_args[0]
+        return compiler.find_library_method(func_args[1:], func_kwargs)
+
+    def _do_dependency(self, name, display_name, kwargs, func_args, func_kwargs):
+        name = display_name = func_args[0]
+        fallbacks = DependencyFallbacksHolder(self.interpreter)
+        varname = func_kwargs.get('variable')
```

Comment:
__[Flake8]__ local variable 'varname' is assigned to but never used ([see on Sider](https://sider.review/gh/repos/19784232/pulls/4595#issue-4884434))

<details>

| Rule |
| - |
| F841 |

</details>
<sub>posted by Sider</sub>